content/frontend/ember-apollo/5-authentication.md

```diff
@@ -0,0 +1,691 @@
+
+```bash
+graphcool pull
+```
+
+</Instruction>
+
+> Note: Before the remote schema gets fetched, you will be asked to confirm that you want to override the current project file. You can confirm by typing `y`. 
+
+This will bump the schema `version` and update the `User` type to now also include the `email` and `password` fields:
+
+```graphql(nocopy)
+type User implements Node {
+  createdAt: DateTime!
+  email: String @isUnique
+  id: ID! @isUnique
+  password: String
+  updatedAt: DateTime!
+}
+```
+
+Next you need to make one more modification to the schema. Generally, when updating the schema of a Graphcool project, you’ve got two ways of doing so:
+
+1. Use the web-based [Graphcool Console](https://console.graph.cool/) and change the schema directly
+2. Use the Graphcool project file and the CLI to update the schema from your local machine
+
+<Instruction>
+
+Open your project file `project.graphcool` and update the `User` and `Link` types as follows:
+
+```graphql
+type Link implements Node {
+  createdAt: DateTime!
+  description: String!
+  id: ID! @isUnique
+  updatedAt: DateTime!
+  url: String!
+  postedBy: User @relation(name: "UsersLinks")
+}
+
+type User implements Node {
+  createdAt: DateTime!
+  id: ID! @isUnique
+  email: String @isUnique
+  updatedAt: DateTime!
+  name: String!
+  password: String
+  links: [Link!]! @relation(name: "UsersLinks")
+}
+```
+
+</Instruction>
+
+You added two things to the schema:
+
+* A new field on the `User` type to store the `name` of the user.
+* A new relation between the `User` and the `Link` type that represents a one-to-many relationship and expresses that one `User` can be associated with multiple links. The relation manifests itself in the two fields `postedBy` and `links`.
+
+<Instruction>
+
+Save the file and execute the following command in the Terminal:
+
+```bash
+graphcool push
+```
+
+</Instruction>
+
+Here is the Terminal output after you execute the command:
+
+```bash(nocopy)
+$ graphcool push
+ ✔ Your schema was successfully updated. Here are the changes: 
+
+  | (*)  The type `User` is updated.
+  ├── (+)  A new field with the name `name` and type `String!` is created.
+  |
+  | (+)  The relation `UsersLinks` is created. It connects the type `Link` with the type `User`.
+
+Your project file project.graphcool was updated. Reload it in your editor if needed.
+```
+
+> **Note**: You can also use the `graphcool status` command after having made changes to the schema to preview the potential changes that would be performed with `graphcool push`.
+
+Perfect, you’re all set now to actually implement the authentication functionality inside your app.
+
+### Authentication with Graph.cool
+
+Time to quickly add the remaning methods to your `auth` service that would allow us to login, sign up, and logout. 
+
+First you need to add the `signInUserMutation` and `createUser` mutations to your mutations folder.
+
+<Instruction>
```

Comment:
this whole instruction block isn't rendered properly. this is an issue with our markdown parser that we currently have to live with unfortunately. the limitation is that an instruction block can at most contain one code block, and this code block must be at the _very end_ of the instruction block. since you're containing multiple code blocks in the same instruction block the rendering doesn't work. the solution is to put create several instruction blocks that each end with the corresponding code block.

Reply:
![image](https://user-images.githubusercontent.com/4058327/29871468-22b5a89a-8d8c-11e7-81bd-c6c2db89d119.png)

Reply:
@nikolasburk I've updated all of the changes. For some reason, it says this isn't outdated, but I believe I fixed it all. I'll take a peek at the Netlify preview, and I'm sure you will check it all also.